.github/workflows/ci_pr.yml

```diff
@@ -106,33 +37,69 @@ jobs:
           - python-version: "3.10"
           - python-version: "3.11"
 
-    name: "Python ${{ matrix.python-version }} Unit Tests"
-    runs-on: ubuntu-20.04
-
-    env:
-      NOSEOPTS: "--with-timer ${{ matrix.additional-nose-opts }}"
-
     steps:
-
-    - name: Checkout WALinuxAgent repo
+    - name: Checkout WALinuxAgent
       uses: actions/checkout@v3
-
+    #
+    # We either install Python and the test dependencies, or download a pre-built virtual environment, depending on the
+    # use_virtual_environment flag.
+    #
     - name: Setup Python ${{ matrix.python-version }}
+      if: (!matrix.use_virtual_environment)
```

Comment:
the workflow parser has this funny behaviour that seems to require parenthesis in order to use the ! operator